### PR TITLE
RFC Mongo2 [reinforce MongoDB on most distros, e.g. for Sugarizer]

### PR DESCRIPTION
**Pull request by jvonau**
### Fixes bug:
mongodb installs on Debian-arm64
### Description of changes proposed in this pull request:
use upstream mongodb repo that provides the latest 4.4 version .
### Smoke-tested on which OS or OS's:
none as of yet

### CODE REVIEW

**Comment by jvonau** [comment]
tested on .58 RaspOS-64bit works ok

File: roles/mongodb/tasks/main2.yml [hunk 1]
```diff
@@ -2,18 +2,19 @@
...
 
-
-- name: Enable & Restart 'mongodb' systemd service if mongodb_enabled, incl daemon-reload (in case mongodb.service changed?)
+# daemon_reload is used to force systemd to reconize a newly installed .service file
+- name: Restart 'mongodb' systemd service if mongodb_enabled
   systemd:
     name: mongodb
     daemon_reload: yes
-    enabled: yes
+    enabled: no
```
Comment (holta) [inline roles/mongodb/tasks/main2.yml]: Why is MongoDB being disabled here?

(The `when: mongodb_enabled | bool` condition below implies the opposite.)

File: roles/mongodb/tasks/main2.yml [hunk 2]
```diff
@@ -2,18 +2,19 @@
   include_tasks: install.yml
   when: mongodb_installed is undefined
 
-
-- name: Enable & Restart 'mongodb' systemd service if mongodb_enabled, incl daemon-reload (in case mongodb.service changed?)
+# daemon_reload is used to force systemd to reconize a newly installed .service file
```
Comment (holta) [inline roles/mongodb/tasks/main2.yml]:
```suggestion
# daemon_reload is used to force systemd to recognize a newly installed .service file
```

File: roles/mongodb/tasks/main.yml [hunk 3]
```diff
@@ -31,18 +31,18 @@
...
+#  when: is_debian_10 and (ansible_architecture == "aarch64")
+#  #when: (is_debian and not is_raspbian) and (not is_debian_8) and (not is_debian_9)    # Test for Debian 10+
+#  ignore_errors: yes
 
 # ELSE...
 
 - name: Install/Enable/Disable/Record MongoDB (main2.yml) if is_raspbian or not is_debian, i.e. not True Debian
   include_tasks: main2.yml
-  when: is_raspbian or not is_debian
+#  when: is_raspbian or not is_debian
```
Comment (holta) [inline roles/mongodb/tasks/main.yml]: Many of the `- name:` descriptions in mongodb/tasks/install.yml (and just above here in mongodb/tasks/main.yml) have residual mentions of `raspbian` conditions...that should be removed now presumably?

Comment (jvonau) [reply roles/mongodb/tasks/main.yml]: Should main.yml and main2.yml be combined again as mongodb now can be installed on all platforms?

Comment (holta) [reply roles/mongodb/tasks/main.yml]: Seems reasonable.  If we're pretty sure MongoDB can be installed easily on most all platforms incl vanilla Debian 10?

File: roles/mongodb/tasks/main2.yml (hunk 1 above)
Comment (jvonau) [reply roles/mongodb/tasks/main2.yml]: moved the stanza

File: roles/mongodb/tasks/install.yml [hunk 4]
```diff
@@ -114,18 +120,25 @@
...
     - { src: 'mongodb.service.j2', dest: '/etc/systemd/system/mongodb.service', mode: '0644' }
     - { src: 'iiab-mongodb-repair-if-no-lock.j2', dest: '/usr/bin/iiab-mongodb-repair-if-no-lock', mode: '0755' }
 
+# daemon_reload is used to force systemd to recognize a newly installed .service file
+# restarted here to ensure the port has been changed
+- name: Disable 'mongodb' systemd service - started on demand by sugarizer
+  systemd:
+    name: mongodb
+    daemon_reload: yes
+    enabled: no
```
Comment (holta) [inline roles/mongodb/tasks/install.yml:145]: Commenting out Line 140 could allow other (non-Sugarizer) services to enable MongoDB as needed.

(If Sugarizer truly wants/needs MongoDB disabled, making that happen explicitly in sugarizer/tasks/install.yml might be best?)

Comment (jvonau) [reply roles/mongodb/tasks/install.yml:145]: I would suggest should new apps appear that the method used in sugarizer's service file be use:
\# Forces mongodb to start up if not started
Requires=mongodb.service
\# Wait for mongodb service to run first
After=mongodb.service

Should these new apps not require a service file then one can add the enabled stanza in their own role.

File: roles/mongodb/tasks/install.yml [hunk 5]
```diff
@@ -18,86 +10,100 @@
...
+
+  - name: Create Linux user mongodb (raspbian)
+    user:
+      name: mongodb
+      group: mongodb    # primary group
+      groups: mongodb
+      home: /var/lib/mongodb
+      shell: /usr/sbin/nologin
+
+  - name: Install /etc/mongod.conf from template
```
Comment (holta) [inline roles/mongodb/tasks/install.yml]: Is installing `/etc/mongod.conf` from template not needed on 64-bit OS's?

As Line 62's end-of-block seems to imply?

(If so, any idea why?)

File: roles/mongodb/tasks/install.yml [hunk 6]
```diff
@@ -18,86 +10,100 @@
...
+
+  - name: Install (move) its 9 TOOLS binaries from /opt/iiab/downloads/mongodb-3.0.1x/tools to /usr/bin (raspbian)
+    shell: mv /tmp/mongodb-3.0.1x/tools/* /usr/bin
+
+  - name: Create Linux group mongodb (raspbian)
+    group:
+      name: mongodb
+      state: present
+
+  - name: Create Linux user mongodb (raspbian)
```
Comment (holta) [inline roles/mongodb/tasks/install.yml]:
```suggestion
  - name: Create Linux user mongodb (32-bit RaspiOS)
```

File: roles/mongodb/tasks/install.yml [hunk 7]
```diff
@@ -18,86 +10,100 @@
...
+  - name: Download & unzip 15MB http://download.iiab.io/packages/mongodb_stretch_3_0_14_tools.zip [IN FACT THIS ONE'S 3.0.15] to /tmp/mongodb-3.0.1x (raspbian)
+    unarchive:
+      remote_src: yes
+      src: "{{ iiab_download_url }}/mongodb_stretch_3_0_14_tools.zip"
+      dest: /tmp/mongodb-3.0.1x
+
+  - name: Install (move) its 9 TOOLS binaries from /opt/iiab/downloads/mongodb-3.0.1x/tools to /usr/bin (raspbian)
+    shell: mv /tmp/mongodb-3.0.1x/tools/* /usr/bin
+
+  - name: Create Linux group mongodb (raspbian)
```
Comment (holta) [inline roles/mongodb/tasks/install.yml]:
```suggestion
  - name: Create Linux group mongodb (32-bit RaspiOS)
```

File: roles/mongodb/tasks/install.yml [hunk 8]
```diff
@@ -18,86 +10,100 @@
...
+  - name: Install (move) its 3 CORE binaries from /tmp/mongodb-3.0.1x/core to /usr/bin (raspbian)
+    shell: mv /tmp/mongodb-3.0.1x/core/* /usr/bin
+
+  - name: Download & unzip 15MB http://download.iiab.io/packages/mongodb_stretch_3_0_14_tools.zip [IN FACT THIS ONE'S 3.0.15] to /tmp/mongodb-3.0.1x (raspbian)
+    unarchive:
+      remote_src: yes
+      src: "{{ iiab_download_url }}/mongodb_stretch_3_0_14_tools.zip"
+      dest: /tmp/mongodb-3.0.1x
+
+  - name: Install (move) its 9 TOOLS binaries from /opt/iiab/downloads/mongodb-3.0.1x/tools to /usr/bin (raspbian)
```
Comment (holta) [inline roles/mongodb/tasks/install.yml]:
```suggestion
  - name: Install (move) its 9 TOOLS binaries from /opt/iiab/downloads/mongodb-3.0.1x/tools to /usr/bin (32-bit RaspiOS)
```

File: roles/mongodb/tasks/install.yml [hunk 9]
```diff
@@ -18,86 +10,100 @@
...
+  - name: Download & unzip 20MB http://download.iiab.io/packages/mongodb_stretch_3_0_14_core.zip to /tmp/mongodb-3.0.1x (raspbian)
+    unarchive:
+      remote_src: yes
+      src: "{{ iiab_download_url }}/mongodb_stretch_3_0_14_core.zip"
+      dest: /tmp/mongodb-3.0.1x
+
+  - name: Install (move) its 3 CORE binaries from /tmp/mongodb-3.0.1x/core to /usr/bin (raspbian)
+    shell: mv /tmp/mongodb-3.0.1x/core/* /usr/bin
+
+  - name: Download & unzip 15MB http://download.iiab.io/packages/mongodb_stretch_3_0_14_tools.zip [IN FACT THIS ONE'S 3.0.15] to /tmp/mongodb-3.0.1x (raspbian)
```
Comment (holta) [inline roles/mongodb/tasks/install.yml]:
```suggestion
  - name: Download & unzip 15MB http://download.iiab.io/packages/mongodb_stretch_3_0_14_tools.zip [IN FACT THIS ONE'S 3.0.15] to /tmp/mongodb-3.0.1x (32-bit RaspiOS)
```

File: roles/mongodb/tasks/install.yml [hunk 10]
```diff
@@ -18,86 +10,100 @@
...
+      path: /tmp/mongodb-3.0.1x
+      state: directory
+
+  - name: Download & unzip 20MB http://download.iiab.io/packages/mongodb_stretch_3_0_14_core.zip to /tmp/mongodb-3.0.1x (raspbian)
+    unarchive:
+      remote_src: yes
+      src: "{{ iiab_download_url }}/mongodb_stretch_3_0_14_core.zip"
+      dest: /tmp/mongodb-3.0.1x
+
+  - name: Install (move) its 3 CORE binaries from /tmp/mongodb-3.0.1x/core to /usr/bin (raspbian)
```
Comment (holta) [inline roles/mongodb/tasks/install.yml]:
```suggestion
  - name: Install (move) its 3 CORE binaries from /tmp/mongodb-3.0.1x/core to /usr/bin (32-bit RaspiOS)
```

File: roles/mongodb/tasks/install.yml [hunk 11]
```diff
@@ -18,86 +10,100 @@
...
-    home: /var/lib/mongodb
-    shell: /usr/sbin/nologin
-  when: is_raspbian | bool
+- block:
+  - name: Create dir /tmp/mongodb-3.0.1x (raspbian)
+    file:
+      path: /tmp/mongodb-3.0.1x
+      state: directory
+
+  - name: Download & unzip 20MB http://download.iiab.io/packages/mongodb_stretch_3_0_14_core.zip to /tmp/mongodb-3.0.1x (raspbian)
```
Comment (holta) [inline roles/mongodb/tasks/install.yml]:
```suggestion
  - name: Download & unzip 20MB http://download.iiab.io/packages/mongodb_stretch_3_0_14_core.zip to /tmp/mongodb-3.0.1x (32-bit RaspiOS)
```

File: roles/mongodb/tasks/install.yml [hunk 12]
```diff
@@ -18,86 +10,100 @@
...
-- name: Create Linux user mongodb (raspbian)
-  user:
-    name: mongodb
-    group: mongodb    # primary group
-    groups: mongodb
-    home: /var/lib/mongodb
-    shell: /usr/sbin/nologin
-  when: is_raspbian | bool
+- block:
+  - name: Create dir /tmp/mongodb-3.0.1x (raspbian)
```
Comment (holta) [inline roles/mongodb/tasks/install.yml]:
```suggestion
  - name: Create dir /tmp/mongodb-3.0.1x (32-bit RaspiOS)
```

File: roles/mongodb/tasks/install.yml (hunk 5 above)
Comment (jvonau) [reply roles/mongodb/tasks/install.yml]: The format of the file changed, the template breaks mongodb version 4.4

**Comment by jvonau** [comment]
Keep up this nonsense and I'll close this PR, you better understand that I'm not your slave.

**Comment by jvonau** [comment]
I'm not looking to test Debian 10 arm64 so I brought back the warning and lockout of debian-10-aarch64 or do we gamble on a silent victory if it works like raspbian-aarch64 for some random user as feedback is sparse when things just work?

**Comment by holta** [comment]
> Debian 10 arm64

> debian-10-aarch64

I can test when I get home later tonight, so long as I know exactly what OS & HW are best to test on.

**Comment by holta** [comment]
fyi this was fresh-installed onto 182-rpi4-2g-u20041-srv-2487-2493 = 10.8.0.22 (together with #2487) and...

http://10.8.0.22/sugarizer would appear to work fine, using MongoDB 4.4.0 here:

```
root@box:~# mongod --version
db version v4.4.0
Build Info: {
    "version": "4.4.0",
    "gitVersion": "563487e100c4215e2dce98d0af2a6a5a2d67c5cf",
    "openSSLVersion": "OpenSSL 1.1.1f  31 Mar 2020",
    "modules": [],
    "allocator": "tcmalloc",
    "environment": {
        "distmod": "ubuntu2004",
        "distarch": "aarch64",
        "target_arch": "aarch64"
    }
}
```

...rather than version 3.6.9 here:

```
root@box:~# apt -a list mongodb
Listing... Done
mongodb/focal 1:3.6.9+really3.6.8+90~g8e540c0b6d-0ubuntu5 arm64
```

**Comment by holta** [comment]
Being tested on a Debian 10.5 VM at 162-deb10-BIG = 10.8.0.38 which is currently in Stage 3 of ./iiab-install

@jvonau do we need to override anything in /etc/iiab/local_vars.yml or otherwise to force Sugarizer & MongoDB to try to install there?

**Comment by deldesir** [comment]
Mongodb won't install on Debian 11 "bullseye". Repository does not have a release file (yet).

**Comment by holta** [comment]
> Mongodb won't install on Debian 11 "bullseye". Repository does not have a release file (yet).

Thanks @deldesir for pointing this out, so folks can prepare for Debian 11 and derivative OS's (RaspiOS 11 etc) when the time comes, and for all your help here with PR #2582 etc!

**Comment by deldesir** [comment]
A quick workaround while we're waiting for Debian to add the release file would be using Debian 10 "Buster" list. I replaced {{ansible_distribution_release}} with buster in line 77 https://github.com/iiab/iiab/blob/master/roles/mongodb/tasks/install.yml#L77.

**Pull request merged**